service/matching/matching_engine_test.go

```diff
@@ -2747,6 +2766,376 @@ func (s *matchingEngineSuite) TestUpdatePhysicalTaskQueueGauge_BuildID() {
+	s.addConsumeAllWorkflowTasksNonConcurrently(100, 1, 1)
+}
+
+func (s *matchingEngineSuite) TestAddConsumeWorkflowTasksDBErrors() {
+	s.taskManager.dbError = true
+	s.addConsumeAllWorkflowTasksNonConcurrently(100, 1, 1)
+}
+
+func (s *matchingEngineSuite) TestMultipleWorkersAddConsumeWorkflowTasksNoDBErrors() {
+	s.addConsumeAllWorkflowTasksNonConcurrently(100, 5, 5)
+}
+
+func (s *matchingEngineSuite) TestMultipleWorkersAddConsumeWorkflowTasksDBErrors() {
+	s.taskManager.dbError = true
+	s.addConsumeAllWorkflowTasksNonConcurrently(100, 5, 5)
+}
+
+func (s *matchingEngineSuite) resetBacklogCounter(numWorkers int, taskCount int, rangeSize int) {
+	s.matchingEngine.config.LongPollExpirationInterval = dynamicconfig.GetDurationPropertyFnFilteredByTaskQueue(1 * time.Millisecond)
+	s.matchingEngine.config.UpdateAckInterval = dynamicconfig.GetDurationPropertyFnFilteredByTaskQueue(100 * time.Millisecond)
+
+	workflowType, workflowExecution := s.generateWorkflowExecution()
+	taskQueue, ptq := s.createTQAndPTQForBacklogTests()
+	s.matchingEngine.config.RangeSize = int64(rangeSize)
+
+	s.addWorkflowTasks(false, numWorkers, taskCount, taskQueue, workflowExecution, nil)
+
+	// TaskID of the first task to be added
+	minTaskID, done := s.taskManager.minTaskID(ptq)
+	s.True(done)
+
+	partitionManager, err := s.matchingEngine.getTaskQueuePartitionManagerNoWait(ptq.Partition(), false)
+	s.NoError(err)
+	pgMgr := s.getPhysicalTaskQueueManagerImpl(ptq)
+
+	s.EqualValues(taskCount*numWorkers, s.taskManager.getTaskCount(ptq))
+
+	// Checking the maxReadLevel with the value of task stored in db
+	maxTaskId, _ := s.taskManager.maxTaskID(ptq)
+	s.EqualValues(maxTaskId, pgMgr.backlogMgr.db.maxReadLevel.Load())
+
+	// validating the approximateBacklogCounter
+	s.EqualValues(taskCount*numWorkers, pgMgr.backlogMgr.db.getApproximateBacklogCount())
+
+	// Unloading the PQM
+	s.matchingEngine.unloadTaskQueuePartition(partitionManager)
+
+	// Simulating a TTL'ed task in Cassandra by removing it from the DB
+	// Remove the task from testTaskManager but not from db/AckManager
+
+	// Stopping the backlogManager so that we TTL and the taskReader does not catch this
+	request := &persistence.CompleteTasksLessThanRequest{
+		NamespaceID:        namespaceId,
+		TaskQueueName:      taskQueue.Name,
+		TaskType:           enumspb.TASK_QUEUE_TYPE_WORKFLOW,
+		ExclusiveMaxTaskID: minTaskID + 1,
+		Limit:              100,
+	}
+	_, err = s.taskManager.CompleteTasksLessThan(context.Background(), request)
+	s.NoError(err)
+	s.EqualValues((taskCount*numWorkers)-1, s.taskManager.getTaskCount(ptq))
+
+	// adding pollers which shall also load the fresher version of tqm
+	s.pollWorkflowTasks(false, workflowType, 1, (taskCount*numWorkers)-1, ptq, taskQueue, nil)
+
+	// Updating pgMgr to have the latest pgMgr
+	pgMgr = s.getPhysicalTaskQueueManagerImpl(ptq)
+
+	// overwriting the maxReadLevel since it could have increased if the previous taskWriter was stopped (which would not result in resetting);
```

Comment:
```suggestion
	// Overwrite the maxReadLevel since it could have increased if the previous taskWriter was stopped (which would not result in resetting);
```